homeassistant/components/seitron/climate.py

```diff
@@ -0,0 +1,140 @@
+
+import logging
+from typing import Any
+
+from pyseitron.seitron_thermostat import SeitronThermostat
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_TENTHS, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .api import AsyncSeitronAuth
+from .const import DOMAIN  # seitron
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the seitron climate entities from a config entry."""
+    coordinator: AsyncSeitronAuth = hass.data[DOMAIN][entry.entry_id]
+    # coordinator.data is a SeitronGateway
+    thermostats = []
+    for thermostat in coordinator.data.devices:
+        _LOGGER.info("Adding entity %s", thermostat.fw_ver)
+        thermostats.append(SeitronClimate(coordinator, thermostat))
+    async_add_entities(
+        SeitronClimate(coordinator, thermostat)
+        for thermostat in coordinator.data.devices
+    )
+
+
+class SeitronClimate(CoordinatorEntity, ClimateEntity):
+    """A Seitron IoT thermostat climate entity."""
+
+    _attr_precision = PRECISION_TENTHS
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.AUTO, HVACMode.COOL, HVACMode.OFF]
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_max_temp = 50
+    _attr_min_temp = -10
+    _attr_target_temperature_high = 40
+    _attr_target_temperature_low = 5
+    _attr_target_temperature_step = 0.1
+    _attr_should_poll = True
+    _attr_icon = "mdi:thermostat-box"
+
+    def __init__(self, coordinator, thermostat: SeitronThermostat) -> None:
+        """Initialize an Seitron climate device."""
+        super().__init__(coordinator)
+        self._thermostat = thermostat
+        self._attr_name = thermostat.name
+        self._attr_unique_id = thermostat.gmac
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, thermostat.gmac)},
+            manufacturer=thermostat.manufacturer,
+            model=thermostat.model,
+            name=thermostat.name,
+            sw_version=thermostat.fw_ver,
+        )
+        _LOGGER.info("Climate: ctored %s %s", thermostat.gmac, thermostat.name)
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the temperature unit."""
+        if self._thermostat.farhenheit:
+            return UnitOfTemperature.FAHRENHEIT
+        return UnitOfTemperature.CELSIUS
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        _LOGGER.info("Climate: getting temp")
```

Comment:
All these info's in all files need to be removed.

Reply:
fixed